hail/src/main/scala/is/hail/expr/types/physical/PNDArray.scala

```diff
@@ -14,7 +14,8 @@ final case class PNDArray(elementType: PType, nDims: Int, override val required:
   assert(elementType.required, "elementType must be required")
 
   def _asIdent: String = s"ndarray_of_${elementType.asIdent}"
-  override def _toPretty = s"NDArray[$elementType,$nDims]"
+
+  override def _toPretty = s"NDArray[${if (elementType.required) "+" else ""}${elementType._toPretty},$nDims]"
```

Comment:
This seems like the wrong fix.

The right one is probably to make a _pretty method like PArray has:

https://github.com/hail-is/hail/blob/f4291ae50e6381806cf75d9d162efad2039ed627/hail/src/main/scala/is/hail/expr/types/physical/PArray.scala#L33-L37

However, this whole system seems a little wonky. PArray shouldn't define _toPretty at all if it overrides _pretty, because that's only called by the _pretty base implementation.

Reply:
Yeah, this is a weird. You have to override `_toPretty` since its abstract on `PType`, but the thing you really want is `_pretty` since it holds onto a `StringBuilder`. I guess for now I'll override `_pretty` and just have `_toPretty` throw an error since it's probably not used anywhere?

Reply:
yeah, I kinda think _toPretty shouldn't exist.